### ISSUE TITLE
Unable to easily add a new field to API response

### ISSUE THREAD

**Issue by noelmccrory** [specmatic/specmatic#1207]
I have an existing API Specification being used by Specmatic for Contract Tests which use examples defined in the API Spec. I am looking to add a new field to the response of one of the endpoints, so my plan was the following:

1. Update the model schema in the API Spec with the new field:

```
        newField:
          type: string
          readOnly: true
```

2. Update Provider Service to return the new field
3. Update Consumers to consume the new field

After completing Step 1, the Contract Tests on the Provider Service continue to pass on master (good). However, on my feature branch where I am updating the Provider Service to return the new field, some of the Contract Tests fail with: 

```
  	  >> RESPONSE.BODY.newField
	  
  	     Value mismatch: key newField in value was unexpected
```

This seems to be because the example response defined in the API Spec does not contain `newField`. If I add `newField` to the example response, then the Contract Tests in my feature branch pass, but the tests on `master` fail with:

```
  	  >> RESPONSE.BODY.newField
	  
  	     Value mismatch: key newfield was missing
```

Ideally I would like to be able to complete Steps 1 and 2 independently, but based on the above, it seems like I would have to update the API Specification and my Provider Service at the same time? Or alternatively create a new version of my API Specification, which doesn't feel right since this isn't a breaking change.

Am I missing something here? Thanks.

System Information:

OS & version: MacOS 14.5
Specmatic version: 1.3.39
JDK version: 21

**Comment by yogeshnikam671** [comment]
Thanks for reporting this. We will get back to you shortly.

**Comment by harikrishnan83** [comment]
Hello @noelmccrory, this is resolved in [release 2.0.7](https://github.com/znsio/specmatic/releases/tag/2.0.7). Please check. Closing this for now, please feel free to re-open if you have any questions on this. Thanks.
